sound/soc/sof/core.c

```diff
@@ -233,8 +233,14 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 
 	ret = snd_sof_machine_register(sdev, plat_data);
 	if (ret < 0)
-		goto fw_trace_err;
+		dev_err(sdev->dev,
+			"error: failed to register machine driver %d\n", ret);
```

Comment:
I think we have conflicting requirements here. On errors, the guidance is to try and clean-up resources, making sure that nothing remains powered. That's what was discussed between GregKH and Takashi recently, so here we should make this configurable via a kernel parameter.